src/charm.py

```diff
@@ -141,22 +143,46 @@ def _on_reset_instance_action(self, event: ActionEvent) -> None:
         event.set_results(results)  # type: ignore[arg-type]
         self.model.unit.status = ops.ActiveStatus()
 
+    def _get_random_password(self) -> str:
+        """Get random password. Extracted from postgresql-k8s charm.
+
+        Returns:
+            random password.
+        """
+        choices = string.ascii_letters + string.digits
+        password = "".join([secrets.choice(choices) for i in range(16)])
+        return password
+
     def _on_register_user_action(self, event: ActionEvent) -> None:
         """Reset instance and report action result.
 
         Args:
             event: Event triggering the reset instance action.
         """
-        results = {
-            "register-user": False,
+        results = {"register-user": False, "user-password": ""}
+        container = self.unit.get_container(SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.MaintenanceStatus("Waiting for pebble")
+            return
+        registration_shared_secret = self._synapse.get_configuration_field(
+            container=container, fieldname="registration_shared_secret"
+        )
+        if registration_shared_secret is None:
+            event.fail("registration_shared_secret was not found")
+            return
+        user_data = {
+            "username": event.params["username"],
+            "admin": event.params["admin"],
+            "password": self._get_random_password(),
         }
-        username = event.params["username"]
-        password = event.params["password"]
-        admin = bool(event.params["admin"] == "yes")
+        user = User(**user_data)
```

Comment:
What if there is a validation error? I'm guessing we're using pydantic here for validation

Reply:
Ops, sure, changed!